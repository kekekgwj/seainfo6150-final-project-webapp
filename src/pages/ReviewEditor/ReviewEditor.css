.review-editor-wrapper {
    width: 1000px;
    margin:  100px auto;
    position: relative;
}
.review-editor-rating {
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    left: -50px;
}
.review-editor-rating-label {
    padding-right: 10px;
}
.review-editor-form {
    margin-top: 50px;
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
}

.review-editor-form input {
    width: 600px;
}

.editor-input-title {
    font-size: 28px;
    font-weight: 600;
    border: 0;
    overflow: hidden;
    height: 30px;
}
.review-editor-body {
    margin-top: 40px;
    border-top: 1px #ddd solid;
    padding-top: 20px;
}
.editor-input-body {
    padding-top: 20px;
    font-size: 16px;
    font-weight: 300;
    overflow: hidden;
    height: 30px;
    width: 600px;
    min-height: 200px;
    max-height: 600px;
    border-style: dotted;
}
.review-editor-submit {
    margin-top: 20px;
    position: absolute;
    right: 200px;
    font-size: 24px;
    background: orange;
    border: 0;
    border-radius: 5px;
}
.review-editor-error {

    font-size: 13px;
    color:red;
}
.review-extra {
    margin-top: 30px;
    width: 600px;
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    font-size: 13px;
}

.review-extra input,select {
    display: inline-block;
    width: 200px;
    border: 1px solid rgb(209, 214, 223);
    padding: 0px 16px;
    border-radius: 4px;
    transition: all 0.3s ease 0s;
    font-family: Roboto, Arial, sans-serif;
    color: rgb(46, 50, 56);
    font-weight: 400;
    font-size: 14px;

}
.review-extra select {
    width: 50px;
}
.review-extra label {
    display: inline-block;
    padding-right: 30px;

}

.review-extra div {
    margin-top: 10px;
}

