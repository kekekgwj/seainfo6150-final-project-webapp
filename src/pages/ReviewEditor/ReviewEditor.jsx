import React, { useState} from "react";
import './ReviewEditor.css';
import Rating from "../../components/Rating/Rating"

import { useHistory, useLocation } from 'react-router-dom';
import { checkSession, sendReviw } from "../../services";
import { errMsgs} from "../../base";

const ReviewEditor = () => {
    const [title, setTitle] = useState('');
    const [body ,setBody] = useState('');
    const [inputValid, setInputValid] = useState(true);
    const [status, setStatus] = useState('');
    const bodyMaxLength = 300;
    const titleMaxLength = 100;

    let score = -1;
    const history = useHistory();
    const location = useLocation();
    
    const getMovieId = () => {
        if (location && location.state) {
            return location.state.id;
        }
       
    }
    const submitReview = () => {

        if (!title) {
            setStatus('title is empty!')
            setInputValid(false);
            return;
        }
        if (title.length > titleMaxLength) {
            setStatus('maximum title length exceeded');
            setInputValid(false);
            return;
        }
        if (!body) {
            setStatus('review is empty');
            setInputValid(false);
            return;
        }
        if (score < 0) {
            setStatus('rating is empty');
            setInputValid(false);
            return;
        } 
        checkSession()
        .then(() => {
            onSendReview();
        })
        .catch((err) => {
            setStatus(errMsgs[err.error] || err.error);
            setTimeout(() => {toLoginPage()}, 1500)
        })
    }
    const getRatingScore = () => {
        const checkedInput = document.querySelector("input[name=rate]:checked");
        
        score = (checkedInput ?  5 - Number(checkedInput.value) + 1: 0) * 2;
    }
    const toLoginPage = () => {
        history.replace({pathname: '/login', 
        state: {nextPathname: `/new_review/${getMovieId()}`, pageInfo: {id: location.state.id,name: location.state.name}}});
    }
    const onSendReview = () => {
        const newReview = {
            title,
            content: body,
            score,
            movieId: location.state.id,
            movieName: location.state.name,
        }
        sendReviw(newReview)
        .then(() => {
            setStatus('Review has been sent');
            setTimeout(()=> {
                history.push(`/movie/${location.state.id}`)
            }, 1500)
        })
        .catch((err) => {
            setStatus(errMsgs[err.error] || err.error);
        })
    }
    return (
        <div className="review-editor-container">
            <div className="review-editor-wrapper">
                <div className="review-editor-snnipet"></div>
                <div className="review-editor-rating">
                    <span className="review-editor-rating-label">Rating this movie:</span>
                    <Rating getRatingScore={getRatingScore}/>
                </div>
                <div className="review-editor-form">
                    <div className="review-editor-title">
                        <input id="title" name="title" 
                            type="text" className="editor-input-title" 
                            placeholder="Type Title"
                            value={title} onChange={e => setTitle(e.target.value)}>
                        </input>
                    </div>
                    <div className="review-editor-body">
                        <textarea id="body" name="body" 
                            type="text" className="editor-input-body" 
                            placeholder="Type review..."
                            maxLength={bodyMaxLength}
                            value={body} onChange={e => setBody(e.target.value)}>
                        </textarea>
                    </div>
                    {!inputValid &&
                    <div className="review-editor-error">
                        <span className="error-message">{status}</span>
                    </div>}
                    <div className="review-extra">
                        <div className="review-extra-tag">
                                <label for="tag">Tag this movie: </label>
                                <input name="tag" id="tag"></input>
                        </div>
                        <div className="review-extra-feel">
                                <label for="tag">Share your feel:</label>
                                <input name="feel" id="feel"></input>
                        </div>
                        <div className="review-extra-recommend">
                            <label for="recommend">Would you recommend it to your friend?</label>
                            <select name="recommend" id="recommend">
                                <option value="yes">YES</option>
                                <option value="no">NO</option>
                            </select>
                        </div>
                        <div className="review-extra-place">
                            <label for="place">Where watch this?</label>
                            <select name="place" id="place">
                                <option value="cinema">CINEMA</option>
                                <option value="home">HOME</option>
                                <option value="school">SCHOOL</option>
                            </select>
                        </div>
                    </div>
                </div>
                <button className="review-editor-submit" onClick={submitReview}>
                    SUBMIT
                </button>
            </div>
        </div>)
    
}

export default ReviewEditor;