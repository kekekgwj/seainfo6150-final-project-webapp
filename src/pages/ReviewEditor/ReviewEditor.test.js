import React from "react";
import ReviewEditor from "./ReviewEditor";

describe("Review page tests", () => {
  it("renders correctly", () => {
    const { container } = render(<ReviewEditor/>);
    expect(container).toMatchSnapshot();
  });
});