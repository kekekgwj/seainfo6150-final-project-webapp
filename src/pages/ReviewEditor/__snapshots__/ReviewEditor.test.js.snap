// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review page tests renders correctly 1`] = `
<div>
  <div
    class="review-editor-container"
  >
    <div
      class="review-editor-wrapper"
    >
      <div
        class="review-editor-snnipet"
      />
      <div
        class="review-editor-rating"
      >
        <span
          class="review-editor-rating-label"
        >
          Rating this movie:
        </span>
        <div
          class="rate-container"
        >
          <div
            class="rate-content"
          >
            <input
              name="rate"
              type="radio"
              value="1"
            />
            <input
              name="rate"
              type="radio"
              value="2"
            />
            <input
              name="rate"
              type="radio"
              value="3"
            />
            <input
              name="rate"
              type="radio"
              value="4"
            />
            <input
              name="rate"
              type="radio"
              value="5"
            />
          </div>
        </div>
      </div>
      <div
        class="review-editor-form"
      >
        <div
          class="review-editor-title"
        >
          <input
            class="editor-input-title"
            id="title"
            name="title"
            placeholder="Type Title"
            type="text"
            value=""
          />
        </div>
        <div
          class="review-editor-body"
        >
          <textarea
            class="editor-input-body"
            id="body"
            maxlength="300"
            name="body"
            placeholder="Type review..."
            type="text"
          />
        </div>
        <div
          class="review-extra"
        >
          <div
            class="review-extra-tag"
          >
            <label
              for="tag"
            >
              Tag this movie: 
            </label>
            <input
              id="tag"
              name="tag"
            />
          </div>
          <div
            class="review-extra-feel"
          >
            <label
              for="tag"
            >
              Share your feel:
            </label>
            <input
              id="feel"
              name="feel"
            />
          </div>
          <div
            class="review-extra-recommend"
          >
            <label
              for="recommend"
            >
              Would you recommend it to your friend?
            </label>
            <select
              id="recommend"
              name="recommend"
            >
              <option
                value="yes"
              >
                YES
              </option>
              <option
                value="no"
              >
                NO
              </option>
            </select>
          </div>
          <div
            class="review-extra-place"
          >
            <label
              for="place"
            >
              Where watch this?
            </label>
            <select
              id="place"
              name="place"
            >
              <option
                value="cinema"
              >
                CINEMA
              </option>
              <option
                value="home"
              >
                HOME
              </option>
              <option
                value="school"
              >
                SCHOOL
              </option>
            </select>
          </div>
        </div>
      </div>
      <button
        class="review-editor-submit"
      >
        SUBMIT
      </button>
    </div>
  </div>
</div>
`;
